apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-focused-launch.tsx

```diff
@@ -3,40 +3,56 @@
  */
 import * as React from 'react';
 import { registerPlugin as originalRegisterPlugin, PluginSettings } from '@wordpress/plugins';
-import { useSelect, useDispatch } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
+import { doAction, hasAction } from '@wordpress/hooks';
+import { addQueryArgs } from '@wordpress/url';
 import FocusedLaunchModal from '@automattic/launch';
 
 /**
  * Internal dependencies
  */
 import { LAUNCH_STORE } from './stores';
+import { redirectParentWindow, redirectToWpcomPath } from './utils';
+
+const HOOK_OPEN_CHECKOUT_MODAL = 'a8c.wpcom-block-editor.openCheckoutModal';
 
 const registerPlugin = ( name: string, settings: Omit< PluginSettings, 'icon' > ) =>
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any
 	originalRegisterPlugin( name, settings as any );
 
 registerPlugin( 'a8c-editor-editor-focused-launch', {
 	render: function LaunchSidebar() {
+		const currentSiteId = window._currentSiteId;
+
 		const isFocusedLaunchOpen = useSelect( ( select ) =>
 			select( LAUNCH_STORE ).isFocusedLaunchOpen()
 		);
 
-		const { closeFocusedLaunch } = useDispatch( LAUNCH_STORE );
-
 		if ( ! isFocusedLaunchOpen ) {
 			return null;
 		}
 
+		const handleCheckout = ( siteId = currentSiteId, isEcommerce = false ) => {
+			// open checkout modal assuming the cart is already updated
+			if ( hasAction( HOOK_OPEN_CHECKOUT_MODAL ) ) {
+				doAction( HOOK_OPEN_CHECKOUT_MODAL );
+				return;
+			}
+
+			// fallback: redirect to /checkout page
+			const checkoutUrl = addQueryArgs( `https://wordpress.com/checkout/${ siteId }`, {
+				// Redirect to My Home after checkout only if the selected plan is not eCommerce
+				...( ! isEcommerce && { redirect_to: `/home/${ siteId }` } ),
+			} );
+			redirectParentWindow( checkoutUrl );
```

Comment:
To further reduce the amount of hardcoded `https://wordpress.com` instances, we could:

```suggestion
			const checkoutUrl = addQueryArgs( `/checkout/${ siteId }`, {
				// Redirect to My Home after checkout only if the selected plan is not eCommerce
				...( ! isEcommerce && { redirect_to: `/home/${ siteId }` } ),
			} );
			redirectToWpcomPath( checkoutUrl );
```

We could also remove the `redirectParentWindow` import